lib/cassandra/protocol/coder.rb

```diff
@@ -370,6 +370,7 @@ def write_value_v3(buffer, value, type)
         when :double           then write_double(buffer, value)
         when :float            then write_float(buffer, value)
         when :int              then write_int(buffer, value)
+        when :smallint         then write_smallint(buffer, value)
```

Comment:
Well, while talking to Cassandra 2.2.4, where I'd created a function and/or aggregate with smallint args (in cqlsh), the Ruby driver had trouble reading the schema afterwards. The decoding error manifested in v3, so I added it there.

This reminds me of a puzzling thing I noticed yesterday: you changed the initcond handling in fetcher from v3 to v4 for Casandra 2.2.x. Based on my error, I presumed 2.2.4 uses v3, implying that your fix for initcond is not right. But if Cassandra 2.2.x uses the v4 protocol, then something else was going wrong in my handshake when the driver was connecting such that it erroneously thought it should be talking v3.

Please clarify.